Firestore/Example/Tests/Integration/API/FIRTimestampAsArgumentTests.m

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2017 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import FirebaseFirestore;
+
+#import <XCTest/XCTest.h>
+
+#import "FirebaseFirestore/FIRTimestamp.h"
+#import "Firestore/Source/Core/FSTFirestoreClient.h"
+
+#import "Firestore/Example/Tests/Util/FSTIntegrationTestCase.h"
+
+@interface FIRTimestampAsArgumentTests : FSTIntegrationTestCase
```

Comment:
Mild nit: depending on how much these overlap with some of the other tests, as much as this arrangement makes it easier to write these tests, it could be harder to find them later when e.g. working on changing queries or user input validation.

To the extent we can do so, we should group these tests with the other tests with similar behavior for the other types. For example query limits are already explored in FIRQueryTests.

However it's also possible that there's just not much overlap, in which case, carry on.